ada/language/ast.py

```diff
@@ -11296,6 +11308,76 @@ def unit_kind():
             ),
         )
 
+    @langkit_property(return_type=T.CompilationUnit.entity.array,
+                      memoized=True, public=True)
+    def withed_units():
+        """
+        Look for all "with" clauses at the top of this compilation unit and
+        return all the compilation units designated by them.
+        """
+        return Self.top_level_with_package_clauses.map(
+            # Try to fetch the compilation unit in a spec file first. If this
+            # fails, the "with" must designate a body without spec (e.g. a
+            # library-level procedure).
+            lambda p: Self.designated_compilation_unit(
+                p.as_symbol_array,
+                kind=UnitSpecification
+            )._or(Self.designated_compilation_unit(
+                p.as_symbol_array,
+                kind=UnitBody
+            )).as_bare_entity
+        )
+
+    @langkit_property(return_type=T.CompilationUnit.entity.array,
+                      memoized=True)
+    def imported_units():
```

Comment:
If you make `withed_units` public, you can probably make this property public too